e2e/solve/test.go

```diff
@@ -40,6 +40,22 @@ type TestOrder struct {
 	RejectReason  string
 }
 
+func (o TestOrder) IsDepositTokenInvalid() bool {
+	return o.Deposit.Token == invalidTokenAddress
```

Comment:
nit: this is very OOP style, rather make these normal functions. In general try to avoid linking state with logic. try decoupling state from logic.

e.g.
```
func isDepositTokenInvalid(o TestOrder) bool {
	return o.Deposit.Token == invalidTokenAddress
}
```